### PR TITLE
Add initial fhir-to-text and nlp route documentation

### PR DESCRIPTION
**Pull request by srcarrier**
Looking to get the *.rst content reviewed prior to generating and re-generating all the html content.

### CODE REVIEW

File: source/routes/fhir-to-text.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+FHIR Resource Text Extraction
+=============================
+This route extracts unstructured data from the following FHIR R4 resources and elements therein:
+- Any FHIR resource containing XHTML narratives at json path $.text.div
+- DocumentReference and DiagnosticReport inline attachments of content type html, pdf, or plain text
+
+Calling the Route
+=================
+This route need not be invoked by another. Simply enable the route, configure your desired nlp service and FHIR resources will be consumed from Kafka, unstructured data within will be extracted and sent to the direct:nlp route to be analyzed.
```
Comment (ccorley) [inline source/routes/fhir-to-text.rst:41]: Slight wordsmithing here: "...from Kafka, then unstructured data within..."

File: source/routes/nlp.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+The LinuxForHealth (LFH) NLP route allows you to send unstructured data to an NLP REST API for analysis. The canonical NLP service response is persisted as a Kafka topic.
+The NLP service employed for analysis is configurable via property settings as shown below. Basic auth is the supported authentication method. Both text/plain and application/json request content types are supported.
+
+Details
+=======
++-------------------------+---------------------------------------------------------------------+
+| Attribute               | Value                                                               |
++=========================+=====================================================================+
+| Route Name              | direct:nlp                                                          |
++-------------------------+---------------------------------------------------------------------+
```
Comment (ccorley) [inline source/routes/nlp.rst:15]: Same comment about route name and route URI.

**Review by ccorley — CHANGES_REQUESTED** [review]
Looks good, just a few minor comments.

File: source/routes/fhir-to-text.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+Unstructured data extracted from FHIR resources is sent to the "direct:nlp" route for processing; see `NLP Route <nlp.html>`_ for details.
+The set of FHIR resource topics consumed from Kafka is configurable as shown below.
+
+Details
+=======
++-------------------------+-----------------------------------------------------------------------------------+
+| Attribute               | Value                                                                             |
++=========================+===================================================================================+
+| Route Name              | "kafka:{{lfh.connect.nlp.fhir-topics}}?brokers={{lfh.connect.datastore.brokers}}" |
++-------------------------+-----------------------------------------------------------------------------------+
```
[the remark this answers is not in the document]
Comment (srcarrier) [reply source/routes/fhir-to-text.rst]: I noticed some inaccuracies w/ the store and notify route doc:
https://linuxforhealth.github.io/docs/routes/direct-storeandnotify.html#details

The route name in the docs is `direct:storeAndNotify`, whereas the route id is `store-and-notify`

I can push a commit to address that ☝️ as part of this pr, if that makes sense.
I included both the route id and uris in my route doc.

File: source/routes/nlp.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+| lfh.connect.nlp.messagetype  | Message type identifier for route (output Kafka topic suffix).            | RESONSE                    |
++------------------------------+---------------------------------------------------------------------------+----------------------------+
+| lfh.connect.nlp.request-json | Json request wrapper. "<REPLACE_TOKEN>" is replaced by the unstructured data at runtime. This is for nlp services that require application/json requests. If plain text is supported, this property may be left blank. | ACD json request wrapper. |
++------------------------------+---------------------------------------------------------------------------+----------------------------+
+
+Sample Annotator for Clinical Data (ACD) properties:
+----------------------------------------------------
+lfh.connect.nlp.uri=https://us-east.wh-acd.cloud.ibm.com/wh-acd/api/v1/analyze/wh_acd.ibm_clinical_insights_v1.0_standard_flow?version=2020-10-22
+lfh.connect.nlp.auth=&authMethod=Basic&authUsername=apikey&authPassword=ENC(<YOUR_ENCRYPTED_PASSWORD_HERE>)
```
Comment (dixonwhitmire) [inline source/routes/nlp.rst:45]: Are we able to utilize basic auth for the NLP route using the camel http4 component params?

**Review by dixonwhitmire — COMMENTED** [review]
Hi @srcarrier lgtm 
I did have a question regarding the basic auth parameters. I just wanted to confirm that they are working correctly with external NLP services since we have had to make adjustments for external FHIR servers.
Thanks!

File: source/routes/fhir-to-text.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (ccorley) [reply source/routes/fhir-to-text.rst]: That would be great, thanks!

**Review by ccorley — APPROVED** [review]
Looks good.  Thanks for the store & notify fix!

**Pull request merged**